code/modules/modular_computers/file_system/programs/robotact.dm

```diff
@@ -56,6 +72,34 @@
 	data["thrustersStatus"] = "[cyborg.ionpulse_on?"ACTIVE":"DISABLED"]" //Feedback for thruster status
 	data["selfDestructAble"] = (cyborg.emagged || istype(cyborg, /mob/living/silicon/robot/model/syndicate))
 
+	data["cyborg_groups"] = list()
+	if(data["masterAI_online"] || (!data["MasterAI_connected"] && cyborg.scrambledcodes)) //If a borg isn't connected, we can just skip this all
+		var/list/borggroup = list() //temporary list for holding groups of borgs
+		for(var/mob/living/silicon/robot/other_borg in GLOB.silicon_mobs)
+			if(!evaluate_borg(cyborg,other_borg))
+				continue
+
+			var/shell = FALSE
+			if(other_borg.shell && !other_borg.ckey)
+				shell = TRUE
+
+			var/list/cyborg_data = list(
+				name = other_borg.name,
```

Comment:
Can you explicify these strings, like "name"? That way we won't have issues if a hypothetical macro with the same lowercased name of one of these keys is defined.

Reply:
I think you've misunderstood me here. What I want is something like this:
```
var/list/data = list(
	"key1" = assoc1,
	"key2" = assoc2,
)
```
instead of 
```
var/list/data = list(
	key1 = assoc1,
	key2 = assoc2,
)
```

Reply:
Like so?

Reply:
mhm